assets/components/textInput/textInput.jsx

```diff
@@ -30,36 +31,45 @@ type PropTypes = {
   onChange: (name: string) => void,
   value: string,
   placeholder: string,
-  required: boolean,
   autocomplete?: AutoComplete,
   autocapitalize?: AutoCapitalize,
-  pattern?: string,
   modifierClasses: Array<?string>,
   onBlur: () => void,
+  showError: boolean,
+  errorMessage: string,
 };
 
 
 // ----- Component ----- //
 
 export default function TextInput(props: PropTypes) {
 
+  const { modifierClasses } = props;
+  if (props.showError) {
+    modifierClasses.push('error');
+  }
+
   return (
-    <div className={classNameWithModifiers('component-text-input', props.modifierClasses)}>
-      <label htmlFor={props.id} className="component-text-input__label">
-        {props.labelText}
-      </label>
-      <input
-        className={classNameWithModifiers('component-text-input__input', props.modifierClasses)}
-        type={props.type}
-        id={props.id}
-        onChange={event => props.onChange(event.target.value || '')}
-        onBlur={props.onBlur}
-        value={props.value}
-        placeholder={props.placeholder}
-        required={props.required}
-        autoComplete={props.autocomplete}
-        autoCapitalize={props.autocapitalize}
-        pattern={props.pattern}
+    <div className="component-text-input__container">
+      <div className={classNameWithModifiers('component-text-input', modifierClasses)}>
```

Comment:
Can `ErrorMessage` live inside this `div` so that you can avoid the container? If not, can the enclosing `div` take `classNameWithModifiers('component-text-input', modifierClasses)`, and the inner `div` be called something like `component-text-input__input-wrapper`?